### PR TITLE
Put QR code reader back

### PR DESCRIPTION
**Pull request by Xelowak**
- [x] Use another compatible QR code scanner
- [x] Fix tests

### CODE REVIEW

File: fe1-web/src/core/components/QrCodeScanner.tsx [hunk 1]
```diff
@@ -54,53 +49,85 @@ const styles = StyleSheet.create({
...
+        <View style={styles.children}>{children}</View>
+      </>
+    );
+  }
+
+  // Scan each code only once
+  let lastValue: string;
+  const onBarCodeScanned = (result: BarCodeScanningResult) => {
+    if (lastValue && lastValue === result.data) return;
+    handleScan(result);
```
Comment (pierluca) [inline fe1-web/src/core/components/QrCodeScanner.tsx:116]: as we had discussed, it makes sense here to just return `result.data` , so that the other APIs don't need to know about `BarCodeScanningResult`, which is an implementation detail from their perspective.

File: fe1-web/src/features/digital-cash/screens/PoPTokenScanner.tsx [hunk 2]
```diff
@@ -70,10 +71,10 @@ const PoPTokenScanner = () => {
     });
   };
 
-  const onScanData = (popToken: string | null) => {
-    if (popToken) {
-      if (tokenMatcher.test(popToken)) {
-        goBack(popToken);
+  const onScanData = ({ data }: BarCodeScanningResult) => {
```
Comment (pierluca) [inline fe1-web/src/features/digital-cash/screens/PoPTokenScanner.tsx:74]: And this would allow us to revert this change too. 
No reason for `PoPTokenScanner` to know the details of how the scanner is implemented.

File: fe1-web/src/features/home/screens/ConnectOpenScan.tsx [hunk 3]
```diff
@@ -70,13 +71,11 @@ const ConnectOpenScan = () => {
     });
   }, [navigation]);
 
-  const handleScan = (data: string | null) => {
+  const handleScan = ({ data }: BarCodeScanningResult) => {
```
Comment (pierluca) [inline fe1-web/src/features/home/screens/ConnectOpenScan.tsx:74]: same

File: fe1-web/src/features/rollCall/screens/RollCallOpened.tsx [hunk 4]
```diff
@@ -157,11 +158,9 @@ const RollCallOpened = () => {
     <>
       <QrCodeScanner
         showCamera={showScanner}
-        handleScan={(data) => {
-          if (data) {
-            addAttendeePopTokenAndShowToast(data);
-          }
-        }}>
+        handleScan={({ data }: BarCodeScanningResult) =>
```
Comment (pierluca) [inline fe1-web/src/features/rollCall/screens/RollCallOpened.tsx:161]: same

**Review by pierluca — CHANGES_REQUESTED** [review]
@Xelowak can you address the Sonar complaints and the last remaining issues ? We're so close to being able to merge this :)

File: fe1-web/src/core/components/QrCodeScanner.tsx [hunk 5]
```diff
@@ -77,30 +78,24 @@ const QrCodeScanner = ({ showCamera, children, handleScan }: IPropTypes) => {
...
-    console.error(err);
-    toast.show(err.toString(), {
-      type: 'danger',
-      placement: 'top',
-      duration: FOUR_SECONDS,
-    });
-  };
-  */
+  if (hasPermission === null) {
+    return <Text>Requesting for camera permission</Text>;
```
[the remark this answers is not in the document]
Comment (Xelowak) [reply fe1-web/src/core/components/QrCodeScanner.tsx]: That's absolutely true, thanks!

**Review by pierluca — APPROVED** [review]
